### PR TITLE
Allow installation with Symfony 7

### PR DESCRIPTION
**Pull request by alexander-schranz**
This will allow the usage of the Symfony Theme Bundle with Symfony 7. There are some blocking dev dependencies which I removed for testing Symfony 7, they are only required for the lint task so we can still make the Tests run on Symfony 7.

Bildschirmfoto 2024-05-08 um 11 14 38

The changed methods should be no bc break as they are all part of final classes.

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -57,13 +65,48 @@ jobs:
                 run: composer require --dev "twig/twig:${{ matrix.twig }}" --no-update --no-scripts
 
             -
-                name: Install dependencies
-                run: composer update
+                name: Remove analyse dependencies
+                run: |
+                    composer remove vimeo/psalm --no-update
+                    composer remove sylius-labs/coding-standard --no-update
```
Comment (alexander-schranz) [inline .github/workflows/build.yml]: This packages are not required to run tests but for linting on the latest dependencies it is required to upgrade it: https://github.com/SyliusLabs/CodingStandard/pull/60

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -57,13 +65,48 @@ jobs:
...
+                    composer remove rector/rector --no-update
```
Comment (alexander-schranz) [inline .github/workflows/build.yml]: This packages are not required to run tests but for linting on the latest dependencies it is required to upgrade it: https://github.com/Sylius/SyliusThemeBundle/pull/132

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -57,13 +65,48 @@ jobs:
                 run: composer require --dev "twig/twig:${{ matrix.twig }}" --no-update --no-scripts
 
             -
-                name: Install dependencies
-                run: composer update
+                name: Remove analyse dependencies
+                run: |
+                    composer remove vimeo/psalm --no-update
```
Comment (alexander-schranz) [inline .github/workflows/build.yml]: This packages are not required to run tests but for linting on the latest dependencies it is required to upgrade it: https://github.com/Sylius/SyliusThemeBundle/pull/133

**Comment by TheMilek** [comment]
Hi @alexander-schranz!
No worries, just reopened up to trigger the tests 😄

**Comment by alexander-schranz** [comment]
@TheMilek Thank you, I fixed the CI part. The Analyse part should be fixed in the psalm upgrade here #133. If we can merge #132 and #133 I can rebase this one.

**Comment by ajgarlag** [comment]
Hi @TheMilek. Can I help in any way to get this PR merged?

**Comment by alexander-schranz** [comment]
@TheMilek rebased!

**Review by TheMilek — APPROVED** [review]

**Comment by TheMilek** [comment]
Thank you, @alexander-schranz!

**Comment by TheMilek** [comment]
> @TheMilek rebased!

Merged! 🚀 
Sorry for the delay in responding, I'll try to improve that. 😅

**Comment by alexander-schranz** [comment]
@TheMilek No problem thanks for the merge :)

**Comment by alexander-schranz** [comment]
Looks like I missed changes in the ThemeCollector which where supressed by psalm annotation: https://github.com/Sylius/SyliusThemeBundle/pull/134

**Pull request merged**